src/slices/usersSlice.ts

```diff
@@ -0,0 +1,48 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+interface UsersState {
+  loaded: boolean,
+  hasError: boolean,
+  users: User[],
+}
+
+const initialState: UsersState = {
+  loaded: false,
+  hasError: false,
+  users: [],
+};
+
+const usersThunk = createAsyncThunk('users/fetch', () => {
+  return getUsers();
+});
+
+const usersSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {
+    setUsers: (state, action: PayloadAction<User[]>) => {
+      state.users = action.payload;
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(usersThunk.pending, (state) => {
+      state.loaded = false;
+      state.hasError = false;
+    });
+
+    builder.addCase(usersThunk.fulfilled, (state, action) => {
+      state.users = action.payload;
+      state.loaded = true;
+    });
+
+    builder.addCase(usersThunk.rejected, (state) => {
+      state.loaded = false;
+      state.hasError = true;
+    });
+  },
+});
+
+export const { setUsers } = usersSlice.actions;
```

Comment:
you never use it:

![image](https://github.com/mate-academy/react_redux-list-of-posts/assets/86923089/744cc763-22c0-4330-9932-ac0bca9f98bb)

so list of users is empty